### PR TITLE
Product analytics tutorial

### LINKED ISSUE #264: tutorial on computing product analytics
product analytics is a common analysis, we can create a tutorial that shows how to compute growth, retention and other metrics using SQL.

I'm unsure if there is an open dataset that we can use, otherwise we can create fake data

### PR DESCRIPTION
**Pull request by neelasha23**
## Describe your changes
Tutorial on product analytics

## Issue number

Closes #264 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--485.org.readthedocs.build/en/485/

### CODE REVIEW

**Comment by neelasha23** [comment]
Reference PR: https://github.com/ploomber/jupysql/pull/321/files

**Comment by lfunderburk** [comment]
Ok, will review today @neelasha23

File: doc/tutorials/product-analytics.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+|--------|------------|--------|------------|
+| 1      | 3|1      | 3|
+| 1      | 3|2      | 4 |
+ 1      | 3| 3| 5| 
+| 2      | 4 |1|3|
+| 2      | 4 |2|4|
+| 2      | 4 |3|5|
+| 3      | 5 |1|3|
+| 3      | 5 |2|4|
+| 3      | 5 |3|5|
```
Comment (edublancas) [inline doc/tutorials/product-analytics.md:142]: can we remove this sample table and add a cell that produces this output?

File: doc/tutorials/product-analytics.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
++++
+
+Then, with **WHERE c1.month = c2.month - 1**, we filter out the total number of users for subsequential months. The expected output is: 
+| c1.month | c1.total_users| c2.month| c2.total_users|
+|--------|------------|--------|------------|
+| 1      | 3|2      | 4 |
+| 2      | 4 |3|5|
```
Comment (edublancas) [inline doc/tutorials/product-analytics.md:150]: can we remove this sample table and add a cell that produces this output?

File: doc/tutorials/product-analytics.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Product Analytics Using JupySQL
```
Comment (edublancas) [inline doc/tutorials/product-analytics.md:15]:
```suggestion
# Product Analytics
```

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by edublancas** [comment]
please fix merging conflicts

**Review by edublancas — APPROVED** [review]

**Pull request merged**